Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -266,13 +267,12 @@ void ConvertUnits::convertQuickly(API::MatrixWorkspace_sptr outputWS,
     // Only do the full check if the quick one passes
     if (commonBoundaries) {
       // Calculate the new (common) X values
-      MantidVec::iterator iter;
-      for (iter = outputWS->dataX(0).begin(); iter != outputWS->dataX(0).end();
-           ++iter) {
+      for (auto iter = outputWS->mutableX(0).begin();
+           iter != outputWS->x(0).end(); ++iter) {
```

Comment:
Careful! You are mixing iterators for `mutableX` and `x`!

Why not `for(auto &x : outputWS->mutableX(0))`?